### ISSUE TITLE
VerbNoContent should allow status codes to be set not hardcode to 204

### ISSUE THREAD

**Issue by arianvp** [haskell-servant/servant#1532]
e.g. 301, 302, 303 are perfectly valid status codes for a Get with No Content.

**Comment by alpmestan** [comment]
Agreed, want to make the patch? We can still have some default definition(s?) that hardcore the status to 204, but indeed we don't want the general construct to hardcode that.
